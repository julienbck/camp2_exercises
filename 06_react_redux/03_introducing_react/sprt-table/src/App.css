.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}


table {
  color:#FFF;
  background-color:#FFF;
  width: 80%;
  box-shadow: -5px 5px 20px #333;
  margin: 0 auto;
  font-size: 12pt;
}
table th {
  background-color: #0080FF;
  height: 50px;
  padding: 0;
  margin: 0;
  text-transform: uppercase;
}
table th button {
  border-style: none;
  background: none;
  text-transform: uppercase;
  color:#FFF;
  font-size: 12pt;

}
table th button:focus {
  outline: none;

}
table th button:hover {
  background-color: #FFF;
  color:#0080FF;
  width: 100%;
  height: 100%;
  transition: background-color 0.2s, color 0.1s;
}
table a {
    text-decoration: none;
    color: #FFF;
}
table tr {
  background-color: #FFF;
}
table > tbody > tr:hover {
  background-color: #045FB4;
  color:#FFF;
  transition: background-color 0.3s, color 0.3s;
}
table > tbody > tr {
    color: #045FB4;
}
table td {
  padding: 10px;
}
.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
