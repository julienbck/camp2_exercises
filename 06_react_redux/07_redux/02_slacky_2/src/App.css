html, body, #root {
  height: 100%;
}

.App {
  text-align: center;
  display:flex;
  flex-direction: column;
  justify-content: flex-start;
  height: 100%;
  align-items: stretch;
  align-content: stretch;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #1e3b5F;
  height: auto;
  width: 100%;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}


.Login {
  padding: 20px;
}

.Login div {
  padding: 20px;
}

.Chat {
  flex: 2 1;
  display:flex;
  flex-flow: column;
  justify-content: space-between;
  padding-top: 5px;
  align-items: stretch;
  align-content: stretch;
  max-height: calc(100vh - 50px)
}

.Chat-messages {
  flex: 1 1 auto;
  overflow: auto;
  max-height: calc(100% - 50px)
}

.Chat-form {
  height: 50px;
  background: #ccc;
  display: flex;
}
.Chat-form form {
  display: flex;
  width: 100%;
}

.Chat-form input {
  flex: 1 0;
  outline: none;
}

.Chat-form button {
  height: 100%;
  margin: 0;
  padding: 20px;
  background-color: #007bff;
  color: white;
  border: none;
  -webkit-appearance: none;
  outline: none;
}
.Chat-form button:hover {
  background-color: #FFF;
  color:#007bff;
  transition: background-color 0.2s, color 0.2s;
}

.message-container {
  margin-bottom: 10px;
  width: 100%;
  text-align: left;
}
.subcontainer{
  color: #FFF;
  height: 2vw;
  vertical-align: middle;
  line-height: 2vw;
  background-color: #1e3b5F;
}
.subcontainer2{
  color: #1e3b5F;
  height: auto;
  padding: 10px;
  vertical-align: middle;
  line-height: 2vw;
  background-color: #FCFCFC;
}

.author {
  font-weight: bold;
  padding: 5px 10px;
}
.date {
  display: flex;
  font-size: 14px;
  font-style: italic;
  padding: 5px 10px;
}
