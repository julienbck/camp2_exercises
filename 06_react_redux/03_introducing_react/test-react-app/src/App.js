import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
          <h1 className="App-title">Hello {this.props.test} ! </h1>
    );
  }
}

export default App;
