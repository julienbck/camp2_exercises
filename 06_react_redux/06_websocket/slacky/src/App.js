import React, { Component } from 'react';
import './App.css';
import Chat from './Chat.js';


class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      username: '',
      message: '',
      messages: [],
      user: []
    };
    this.updateUsername = this.updateUsername.bind(this);
    this.updateMessage = this.updateMessage.bind(this);
    this.updateBoardUsername = this.updateBoardUsername.bind(this);
  }
  componentDidMount() {
    // We open a connection from the client, related to the server
    const ws = new WebSocket("ws://localhost:4000");
    const nbUsersElem = document.getElementById("nbUsers");
    // What to do when we receive a message?
    ws.onmessage = (event) => {
      console.log("Message: " + event.data);
      nbUsersElem.innerHTML = event.data;
    };

    // Alert the server that the client is gone
    window.addEventListener("beforeunload", () => ws.send("CLOSE"));
  }
  updateUsername(event){
    this.setState({
      username: event.target.value
    });
  }
  updateMessage(message){
    this.setState({
      message: message.target.value,
      messages: this.state.messages + [message.target.value]
    });
  }
  updateBoardUsername (event){
    let push = this.state.users.push(event.target.value);
    this.setState({
      users: push
    })
  }
  render() {
    return(
      <div id="App">
      <p>There is <span id="nbUsers">0</span> users connected.</p>
        <form>
        <label>
        Username:
        <input type="text" onChange={this.updateUsername} value={this.state.username}></input>
        </label>
        <input type="submit" onClick={this.updateBoardUsername} value="Submit" />
        </form>
        <form>
        <label>
        Message:
        <input type="text" onClik={this.updateMessage} value={this.state.message}></input>
        </label>
        </form>
        <Chat username={this.state.username} message={this.state.message} />
        <h3>{this.state.messages}</h3>

        <h4>{this.state.users}</h4>
      </div>
    );
  }
}



export default App;
