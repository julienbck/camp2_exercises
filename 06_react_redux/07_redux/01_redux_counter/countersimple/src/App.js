import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(props){
    super(props);
    this.state = {
      counter: 0
    }
  }
  increment1 = () => {
    this.setState({
      counter: this.state.counter + 1
    })
  }
  decrement1 = () => {
    this.setState({
      counter: this.state.counter - 1
    })
  }
  increment10 = () => {
    this.setState({
      counter: this.state.counter + 10
    })
  }
  decrement10 = () => {
    this.setState({
      counter: this.state.counter - 10
    })
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
        <button onClick={this.decrement10}> - by 10</button>
          <button onClick={this.decrement1}> - by 1 </button>
          <span>{this.state.counter}</span>
          <button onClick={this.increment1}>+ by 1</button>
          <button onClick={this.increment10}>+ by 10</button>
        </p>
      </div>
    );
  }
}

export default App;
