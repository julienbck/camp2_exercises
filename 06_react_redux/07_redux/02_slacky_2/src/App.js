import React, { Component } from "react";
import "./App.css";
import Login from "./Login";
import Chat from "./Chat";
import { createStore } from 'redux';



class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: null,
      messages: [],
      date: ""
    };
    // Attaching the websocket to our App so we can reuse it
    this.websocket = new WebSocket("ws://localhost:8080");
  }
  componentDidMount() {
    // Listen for messages
    this.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          return;
        case "MESSAGES":
          this.setState({ messages: message.data });
          console.log(message.data);
          return;
      }
    });
  }

  handleUserName = userName => {
    this.setState({ userName: userName });
    this.websocket.send(
      JSON.stringify({
        type: "LOGIN",
        userName: userName
      })
    );
  };

  sendMessage = message => {
    this.websocket.send(
      JSON.stringify({
        type: "NEW_MESSAGE",
        userName: this.state.userName,
        message: message,
      })
    );
  };
  render() {
    return (
      <div className="App">
        <header className="App-header">
            {this.state.userName ? (
            <p>
            Hi {this.state.userName} !</p>
          ) : (
            <p>Welcome
            </p>
          )}
        </header>

        {this.state.userName ? (
          <Chat sendMessage={this.sendMessage} messages={this.state.messages} />
        ) : (
          <Login handleUserName={this.handleUserName} />
        )}
      </div>
    );
  }
}


export default App;
