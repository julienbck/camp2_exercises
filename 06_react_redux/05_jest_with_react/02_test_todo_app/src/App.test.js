import React from 'react';
import App from './App.js';
import renderer from 'react-test-renderer';
import ReactDOM from "react-dom";
import Wapiti from "wapiti";


test("everything looks the same", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test("Check if field input is empty at first launch", () => {
  expect.assertions(1);

  return Wapiti.goto("http://localhost:3000")
    .capture(() => document.querySelector("input").textContent)
    .run()
    .then(result => {
      expect(result).toBe("");
    });
});

test("Check if can add text in field input and if have text in list", () => {
  expect.assertions(1);

  return Wapiti.goto("http://localhost:3000")
    // .fillForm({
    //   ".inputToDo": "test1"
    // })
    .puppeteer(page =>
      page.type(".inputToDo", "test1")
    )
    .click("button")
    .capture(() => document.querySelector("ul li span").textContent)
    .run()
    .then(result => {
      expect(result).toBe("test1");
    });
});
test("Check if checkbox is on ", () => {
  expect.assertions(1);

  return Wapiti.goto("http://localhost:3000")
    // .fillForm({
    //   ".inputToDo": "test1"
    // })
    .puppeteer(page =>
      page.type(".inputToDo", "test1")
    )
    .click("button")
    .click(".checkboxToDisplayDelete")
    .click(".deleteButton")
    .capture(() => document.querySelector("ul li span").textContent)
    .run()
    .then(result => {
      expect(result).toBe("");
    });
});
