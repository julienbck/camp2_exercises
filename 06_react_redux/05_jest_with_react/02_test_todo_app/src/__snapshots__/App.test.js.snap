// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`everything looks the same 1`] = `
<div
  className="Todo"
>
  <div
    className="new-todo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="inputToDo"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    className="todos"
  />
</div>
`;
